File: src/main/java/no/nav/aap/api/søknad/tokenx/TokenXConfig.java
package no.nav.aap.api.søknad.tokenx;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import no.nav.security.token.support.client.core.oauth2.OAuth2AccessTokenResponse;
import org.springframework.boot.autoconfigure.jackson.Jackson2ObjectMapperBuilderCustomizer;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.Optional;

@Configuration
 class TokenXConfig {
    @Bean
     TokenXConfigMatcher configMatcher() {
        return (cfgs, req) -> {
            return Optional.ofNullable(cfgs.getRegistration().get(req.getHost().split("\\.")[0]));
        };
    }

    @Bean
    public Jackson2ObjectMapperBuilderCustomizer customizer() {
        return b -> b.mixIn(OAuth2AccessTokenResponse.class, IgnoreUnknownMixin.class);
    }

    @JsonIgnoreProperties(ignoreUnknown = true)
    private interface IgnoreUnknownMixin {
    }
}

File: src/main/kotlin/no/nav/aap/api/søknad/tokenx/TokenXConfig.kt
package no.nav.aap.api.søknad.tokenx

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import no.nav.security.token.support.client.core.ClientProperties
import no.nav.security.token.support.client.core.oauth2.OAuth2AccessTokenResponse
import no.nav.security.token.support.client.spring.ClientConfigurationProperties
import org.springframework.boot.autoconfigure.jackson.Jackson2ObjectMapperBuilderCustomizer
import org.springframework.context.annotation.Bean
import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder
import java.net.URI

class TokenXConfig {
    @Bean
    fun configMatcher(): TokenXConfigMatcher {
        return CCM()
    }
    @Bean
    fun customizer(): Jackson2ObjectMapperBuilderCustomizer {
        return Jackson2ObjectMapperBuilderCustomizer { b: Jackson2ObjectMapperBuilder ->
            b.mixIn(
                OAuth2AccessTokenResponse::class.java, IgnoreUnknownMixin::class.java
            )
        }
    }

    @JsonIgnoreProperties(ignoreUnknown = true)
    private interface IgnoreUnknownMixin

    internal class CCM : TokenXConfigMatcher {
        override fun findProperties(cfg: ClientConfigurationProperties, uri: URI): ClientProperties? {
            return cfg.registration[uri.host.split("\\.".toRegex()).toTypedArray()[0]]
        }
    }
}